### PR TITLE
Test SlabLand–atmosphere MOST surface-flux coupling

### PR DESCRIPTION
**Pull request by ewquon**
## Summary

Adds test coverage for the `SlabLand` ↔ atmosphere Monin–Obukhov (MOST) turbulent-flux coupling, which had two gaps:

- **Compute side** — the *land* turbulent-flux values were never checked numerically (only the atmosphere–ocean path was, in `test_surface_fluxes.jl`).
- **Apply side** — the computed MOST surface stress reaching the atmosphere's bottom momentum boundary condition (the "atmosphere feels drag" feedback) had **zero** coverage for any surface; the existing Breeze test only exercises `SlabOcean` construction and SST response.

## Tests added

**`test/test_slab_land.jl`**
- *Atmosphere–Land turbulent fluxes (analytic neutral)* — with stability/gustiness neutralized and a constant roughness length, asserts `x_momentum`, `y_momentum`, `friction_velocity`, and `sensible_heat` against the analytic log-law form (the land analog of the atmosphere–ocean analytic test).
- *Atmosphere–Land flux stability and roughness response* — with the **real** land stability functions: sensible-heat sign tracks the surface–air temperature gradient; `u★` orders unstable > neutral > stable; larger roughness increases drag.

**`test/test_breeze_coupling.jl`**
- *Surface stress feedback* — a Breeze atmosphere coupled to a `SlabLand`: the atmosphere's stored `net_fluxes` momentum fields **are** the bottom-flux BC fields (`===` identity), and after `update_state!` those fields equal the computed land surface stress (`x_momentum`/`y_momentum`) and are nonzero.

All tests run on a single column / small grid, 0–1 steps, and are deterministic.

## Verification

Each test's assertions were verified to pass against the coupling running with **Oceananigans#main + Breeze#main** (the versions the coupling actually runs with). The repository's local test environment currently has an unrelated `Manifest` breakage (`KeyError: Zarr`), so the canonical full-suite run is left to CI.

## Not covered (follow-up)

Spatially-varying / `u★`-dependent land roughness (`LandRoughnessLength`) is not wired into `SlabLand` (roughness lives on the flux closure, and the per-cell-field path is unused), so it is intentionally left for a follow-up rather than scaffolding an unsupported path.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: test/test_slab_land.jl [hunk 1]
```diff
@@ -339,3 +342,118 @@ end
...
+            land = SlabLand(grid; hydrology = DryLand(), energy = SlabEnergy(eltype(grid)))
+            set!(land; T = T_skin)
+            model = AtmosphereLandModel(atmosphere, land; atmosphere_land_fluxes = fluxes, radiation = nothing)
+            update_state!(model)
+            f  = model.interfaces.atmosphere_land_interface.fluxes
+            cp = Thermodynamics.cp_m(atmosphere.thermodynamics_parameters, 0.003)
+            g  = model.interfaces.properties.gravitational_acceleration
+            return (u★ = @allowscalar(f.friction_velocity[1, 1, 1]),
+                    Q  = @allowscalar(f.sensible_heat[1, 1, 1]),
+                    neutral_skin = 288 + h / cp * g)
```
Comment (xkykai) [inline test/test_slab_land.jl]:
```suggestion
            return (u★ = @allowscalar(f.friction_velocity[1, 1, 1]),
                    Q  = @allowscalar(f.sensible_heat[1, 1, 1]))
```

File: test/test_slab_land.jl [hunk 2]
```diff
@@ -339,3 +342,118 @@ end
...
+        end
+
+        neutral_skin = land_flux_response(290).neutral_skin
```
Comment (xkykai) [inline test/test_slab_land.jl]:
```suggestion

```

File: test/test_slab_land.jl [hunk 3]
```diff
@@ -339,3 +342,118 @@ end
...
+
+        @test @allowscalar(interface_fluxes.x_momentum[1, 1, 1])        ≈ τˣ
+        @test @allowscalar(interface_fluxes.friction_velocity[1, 1, 1]) ≈ u★
+        @test @allowscalar(interface_fluxes.sensible_heat[1, 1, 1])     ≈ 𝒬ᵀ
+        @test @allowscalar(abs(interface_fluxes.y_momentum[1, 1, 1]))   < eps(Float64)
+    end
+end
+
+@testset "Atmosphere-Land flux stability and roughness response" begin
+    for arch in test_architectures
```
Comment (xkykai) [inline test/test_slab_land.jl:412]:
```suggestion
    for arch in test_architectures
    
        T_atm = 288
        q_atm = 0.003
        cp       = Thermodynamics.cp_m(AtmosphereThermodynamicsParameters(Float64), q_atm)
        g         = 9.80665
        neutral_skin = T_atm + 10 / cp * g
```

File: test/test_slab_land.jl [hunk 4]
```diff
@@ -339,3 +342,118 @@ end
...
+        # the friction velocity and sensible heat for a given skin temperature / roughness.
+        function land_flux_response(T_skin; ℓ = 0.1)
+            grid = LatitudeLongitudeGrid(arch, Float64;
+                                         size = 1, latitude = 10, longitude = 10,
+                                         z = (-1, 0), topology = (Flat, Flat, Bounded))
+            h = 10.0
+            atmosphere = PrescribedAtmosphere(grid; surface_layer_height = h, boundary_layer_height = 512)
+            @allowscalar begin
+                fill!(parent(atmosphere.tracers.T),    288)
+                fill!(parent(atmosphere.tracers.q),    0.003)
```
Comment (xkykai) [inline test/test_slab_land.jl]:
```suggestion
                fill!(parent(atmosphere.tracers.T),    T_atm)
                fill!(parent(atmosphere.tracers.q),    q_atm)
```

File: test/test_slab_land.jl [hunk 5]
```diff
@@ -339,3 +342,118 @@ end
...
+        𝒬ᵀ = - ρᵃᵗ * cp * u★ * θ★
+
+        @test @allowscalar(interface_fluxes.x_momentum[1, 1, 1])        ≈ τˣ
+        @test @allowscalar(interface_fluxes.friction_velocity[1, 1, 1]) ≈ u★
+        @test @allowscalar(interface_fluxes.sensible_heat[1, 1, 1])     ≈ 𝒬ᵀ
+        @test @allowscalar(abs(interface_fluxes.y_momentum[1, 1, 1]))   < eps(Float64)
+    end
+end
+
+@testset "Atmosphere-Land flux stability and roughness response" begin
```
Comment (xkykai) [inline test/test_slab_land.jl:411]: Basically I was just a bit confused by 288 since it's a magic number, so I just make changes to make it more explicit. @ewquon your call!

**Review by xkykai — APPROVED** [review]
some minimal changes to make it more understandable (for me). feel free to incorporate them, rewrite them, or ignore them. I'll approve

File: test/test_slab_land.jl (hunk 5 above)
Comment (ewquon) [reply test/test_slab_land.jl:411]: Good suggestion!
